src/api/index.ts

```diff
@@ -0,0 +1,23 @@
+export { getCurrentSemester } from './api';
+
+export interface Fraction {
+	numerator: number;
+	denominator: number;
+}
+
+export interface CourseItem {
+	name: string;
+	weight: number;
+	grade?: number | Fraction;
```

Comment:
Can you add an optional dueDate here?

Reply:
Also, from calculating the grade last semester, I found that it was difficult to work with grade being type number or Fraction because the two cases had to be dealt with separately and it introduced a lot of unnecessary conditionals.  It might be easier to split this up and have two variables instead, one for marksEarned and the other be marksTotal.  This may have the advantage of possibly making the calculations easier, the data cleaner, and would mean that you no longer need to introduce a Fraction type.

Reply:
I was wondering if we would ever accept a percentage as input and that is why I considered the number, but if it adds a lot of complexity then yeah maybe it isn't worth it. I will change, and we will need to make respective changes in the UI I think.